File: coding/knapsack/subset_sum_recursive.py
def isSubSetSum(set, n, sum):
    if sum == 0:
        return True
    if (n == 0 and sum != 0):
        return False
    if set[n-1] > sum:
        return isSubSetSum(set, n-1, sum)
    return isSubSetSum(set, n-1, sum - set[n-1]) or isSubSetSum(set, n-1, sum)

set = [3, 27, 34, 4, 12, 5, 2]
sum = 30
n = 6

print(isSubSetSum(set, n, sum))

File: coding/knapsack/knapsack_ruby/subset_sum_recursive.rb
set = [3, 27, 34, 4, 12, 5, 2]
sum = 30
n = set.length

# print(isSubSetSum(set, n, sum))

def isSubSetSum(set, sum, n)
  return true if sum == 0
  return false if n == 0 && sum != 0

  if sum < set[n - 1]
    return isSubSetSum(set, sum, n - 1)
  end

  isSubSetSum(set, sum - set[n - 1], n - 1) || isSubSetSum(set, sum, n - 1)
end

p isSubSetSum(set, sum, n)

# >> false
